library/src/main/java/com/devbrackets/android/exomedia/core/video/scale/MatrixManager.java

```diff
@@ -97,7 +97,9 @@ protected void applyCenterCrop(@NonNull TextureView view, @NonNull Matrix transf
         float yScale = (float)view.getHeight() / intrinsicVideoSize.y;
 
         float scale = Math.max(xScale, yScale);
-        applyScale(view, transformMatrix, scale, scale);
+        xScale = scale / xScale;
```

Comment:
On the contrary, we need the max of the width and height ratios to use as a normalizing scale to apply to those ratios before they can be used to scale. If we didn't normalize them they could potentially stretch.

Consider this case:

```java
videoWidth = 720;
videoHeight = 1280;
viewWidth = 1440;
viewHeight = 1446; // I know this is an odd number :-)
```

Using the current code, the scale would be `2.0` for the x and y, which does not match the ratio of the video size, so it gets stretched. Using the proposed change results in a scale of `1` for the x, and `1.7704011065` for the y, which matches the ratio between the video size (slightly correcting for the differences in view size).

I just noticed the same issue in the other scaling calculations. I'm not sure if I'm explaining it correctly, [but this project has done a great job of implementing scale](https://github.com/yqritc/Android-ScalableVideoView/blob/master/library/src/main/java/com/yqritc/scalablevideoview/ScaleManager.java). Perhaps this is better done as part of a separate pull request that fixes all the scale calculations.